tests/test_cgnsutilities.py

```diff
@@ -56,33 +63,45 @@ def test_overwriteBCs(self):
         bcFile = os.path.abspath(os.path.join(baseDir, "../examples/overwriteBCs_bcFile"))
         # Check the BC before overwriting. Note that the "updated" BC is first deleted and new appended
         self.assertEqual(self.grid.blocks[0].bocos[0].family, "wall")
-        self.assertEqual(self.grid.blocks[0].bocos[0].type, BC["bcwallviscous"])
+        self.assertEqual(self.grid.blocks[0].bocos[0].cgnsType, BCSTANDARD["bcwallviscous"])
         self.grid.overwriteBCs(bcFile)
         self.assertEqual(self.grid.blocks[0].bocos[-1].family, "wall_inviscid")
-        self.assertEqual(self.grid.blocks[0].bocos[-1].type, BC["bcwallinviscid"])
+        self.assertEqual(self.grid.blocks[0].bocos[-1].cgnsType, BCSTANDARD["bcwallinviscid"])
 
     def test_overwriteBCfamily(self):
         # Find a specific family and overwrite the BCs for the entire family
         # Check the BC before overwriting
         self.assertEqual(self.grid.blocks[0].bocos[1].family, "Far")
-        self.assertEqual(self.grid.blocks[0].bocos[1].type, BC["bcfarfield"])
+        self.assertEqual(self.grid.blocks[0].bocos[1].cgnsType, BCSTANDARD["bcfarfield"])
         self.assertEqual(self.grid.blocks[1].bocos[1].family, "Far")
-        self.assertEqual(self.grid.blocks[1].bocos[1].type, BC["bcfarfield"])
+        self.assertEqual(self.grid.blocks[1].bocos[1].cgnsType, BCSTANDARD["bcfarfield"])
         self.assertEqual(self.grid.blocks[1].bocos[0].family, "wall")
-        self.assertEqual(self.grid.blocks[1].bocos[0].type, BC["bcwallviscous"])
+        self.assertEqual(self.grid.blocks[1].bocos[0].cgnsType, BCSTANDARD["bcwallviscous"])
         self.grid.overwriteBCFamilyWithBC("Far", "bcoverset", blockIDs=[2])
 
         # block 0 should be unchanged even though family matches
         self.assertEqual(self.grid.blocks[0].bocos[1].family, "Far")
-        self.assertEqual(self.grid.blocks[0].bocos[1].type, BC["bcfarfield"])
+        self.assertEqual(self.grid.blocks[0].bocos[1].cgnsType, BCSTANDARD["bcfarfield"])
 
         # block 1 wall family should be unchanged because family doesn't match
         self.assertEqual(self.grid.blocks[1].bocos[0].family, "wall")
-        self.assertEqual(self.grid.blocks[1].bocos[0].type, BC["bcwallviscous"])
+        self.assertEqual(self.grid.blocks[1].bocos[0].cgnsType, BCSTANDARD["bcwallviscous"])
 
         # block 1 Far family should be overwritten with bcoverset
         self.assertEqual(self.grid.blocks[1].bocos[1].family, "Far")
-        self.assertEqual(self.grid.blocks[1].bocos[1].type, BC["bcoverset"])
+        self.assertEqual(self.grid.blocks[1].bocos[1].cgnsType, CGNSUSERDEFINEDTYPE)
+        self.assertEqual(self.grid.blocks[1].bocos[1].cgnsUserDefined, BCUSERDEFINED["bcoverset"])
+
+        # Test if we can rename families, then overwrite the BC using the family name to a user defined BC
```

Comment:
This section should be its own test, separate from `test_overwriteBCfamily`. It could even be two tests: one for `renameFamilies` and one for the read and write with user defined BCs.